src/Type/WebMozartAssert/AssertTypeSpecifyingExtension.php

```diff
@@ -407,33 +408,37 @@ private function getExpressionResolvers(): array
 				},
 				'isInstanceOf' => static function (Scope $scope, Arg $expr, Arg $class): ?Expr {
 					$classType = $scope->getType($class->value);
-					$classNameStrings = $classType->getConstantStrings();
-					if (count($classNameStrings) !== 1) {
-						$classNames = $classType->getObjectClassNames();
-						if (count($classNames) === 1) {
-							return new Instanceof_(
-								$expr->value,
-								new Name($classNames[0])
-							);
-						}
-						return null;
+					$classNames = $classType->getObjectTypeOrClassStringObjectType()->getObjectClassNames();
```

Comment:
This now means the code will also do something with `Assert::isInstanceOf($value, $object);`

1) Can you test that this actually works in the assert library?
2) Can you write a type inference test for this too?

Reply:
@ondrejmirtes

1. This method uses the `instanceof` operator under the hood, which seems to accept instances of objects as well as class names. I used the following code to verify this:
```php
use Webmozart\Assert\Assert;

$object = new stdClass();
Assert::isInstanceOf($object, new stdClass());
echo 'It works!';
```
2. This test already exists, are there additional cases we should cover?: https://github.com/phpstan/phpstan-webmozart-assert/blob/983a1a7a882daa438a0d37996426f82944f10102/tests/Type/WebMozartAssert/data/type.php#L196-L197